website/docs/secrets/managing-secrets-with-sops.mdx

```diff
@@ -0,0 +1,183 @@
+---
+title: Managing secrets with Sops
+hide_title: true
+---
+
+# Managing secrets with Sops
+
+Weave GitOps Enterprise now supports managing secrets using [SOPS](https://github.com/mozilla/sops), a tool that encrypts and decrypts secrets using various key management services.
+
+In this document, we will cover the prerequisites for using SOPS with Weave GitOps Enterprise, and how to configure SOPS for your kubernetes cluster to work with GPG and age keys.
+
+
+## Encrypting secrets using OpenPGP
+
+OpenPGP is a way of using Sops to encrypt and decrypt secrets with Weave GitOps Enterprise.
+
+Here are the steps to generate an OpenPGP key and configure your cluster to work with Weave GitOps Enterprise secrets management.
+
+1- Generate a gpg key pairs
+
+```bash
+export KEY_NAME="gpg-key"
+export KEY_COMMENT="gpg key"
+
+gpg --batch --full-generate-key <<EOF
+%no-protection
+Key-Type: 1
+Key-Length: 4096
+Subkey-Type: 1
+Subkey-Length: 4096
+Expire-Date: 0
+Name-Comment: ${KEY_COMMENT}
+Name-Real: ${KEY_NAME}
+EOF
+```
+
+2- Export the key pairs fingerprint in the shell
+```bash
+gpg --list-secret-keys "${KEY_NAME}"
+
+sec   rsa4096 2020-09-06 [SC]
+      710DC0DB6C1662F707095FC30233CB21E656A3CB
+      
+export KEY_FP="710DC0DB6C1662F707095FC30233CB21E656A3CB"
+```
+
+3- Export the generated private key to a kubernetes secret `sops-gpg-private-key` which will be used by flux's kustomize-controller to decrypt the secrets using sops.
+
+```bash
+gpg --export-secret-keys --armor "${KEY_FP}" | kubectl create secret generic sops-gpg-private-key --namespace=flux-system --from-file=sops.asc=/dev/stdin
+```
+
+4- Export the generated public key to a kubernetes secret `sops-gpg-public-key` which will be used by Weave GitOps Enterprise to encrypt the secrets created from the UI.
+
+```bash
+gpg --export --armor "${KEY_FP}" | kubectl create secret generic sops-gpg-public-key --namespace=flux-system --from-file=sops.asc=/dev/stdin
+```
+
+> It's recommended to remove the secret from your machine
+
+```bash
+gpg --delete-secret-keys "${KEY_FP}"
+```
+
+5- Create a kustomization for reconciling the secrets on the cluster and set the `--decryption-secret` flag to the name of the private key created in step 3
+
+```bash
+flux create kustomization secrets \
+--source=secrets \
```

Comment:
can you add a comment that this is the gitsoure it will reconcile the secrets from, can be an inline comment